docs/README.md

```diff
@@ -0,0 +1,613 @@
+resource which shall automatically be disposed off and closed properly afterwards.
+
+```scala mdoc:reset
+import java.io.Closeable
+
+import com.velocidi.apso.TryWith
+
+def buildResource = new Closeable {
+  override def toString: String = "good resource"
+  def close(): Unit = {
+    println("Resource is now Closed")
+  }
+}
+
+def goodHandler(resource: Closeable) = {
+  println(resource)
+}
+
+def badHandler(resource: Closeable) = {
+  throw new Exception()
+}
+
+TryWith(buildResource)(goodHandler)
+
+TryWith(buildResource)(badHandler)
+```
+
+## Akka HTTP
+
+The `akka-http` module provides additional [directives](https://doc.akka.io/docs/akka-http/current/routing-dsl/directives/index.html#directives) to be used in [akka-http](https://doc.akka.io/docs/akka-http/current/index.html).
+
+To use it in an existing SBT project, add the following dependency to your `build.sbt`:
+
+```scala
+libraryDependencies += "com.velocidi" %% "apso-akka-http" % "@VERSION@"
+```
+
+### ClientIPDirectives
+
+The `ClientIPDirectives` trait exposes an `optionalRawClientIP` directive that extracts the raw IP of the client from either the `X-Forwarded-For`, `Remote-Address` or `X-Real-IP` header, in that order of priority.
+
+### ExtraMiscDirectives
+
+The `ExtraMiscDirectives` trait exposes the directives `cacheControlMaxAge(maxAgeDuration)` and `optionalRefererHost` to set the cache-control header to the supplied finite duration (the minimum resolution is 1 second) to extract the referer from the HTTP request header, respectively. The `ExtraMiscDirectives` companion object exposes a `cacheControlNoCache` directive to reply with the `no-cache` option in the `Cache-Control` header.
+
+### ProxySupport
+
+The `ProxySupport` traits adds helper methods to proxy requests to a given uri, either directly (`proxyTo`), or with the unmatched path and query parameters of the current context (`proxyToUnmatchedPath`).
+
+## Amazon Web Services
+
+Apso provides a group of classes to ease the interaction with the Amazon Web Services, namely S3 and EC2.
+
+To use it in an existing SBT project, add the following dependency to your `build.sbt`:
+
+```scala
+libraryDependencies += "com.velocidi" %% "apso-aws" % "@VERSION@"
+```
+
+### ConfigCredentialsProvider
+
+The `ConfigCredentialsProvider` is an `AWSCredentialsProvider` (from AWS SDK for Java) that retrieves credentials from a typesafe configuration, allowing customization of its `Config` object, as well as the access key and secret key paths:
+
+```scala mdoc:reset
+import com.velocidi.apso.aws._
+
+import com.typesafe.config._
+
+val confProvider = ConfigCredentialsProvider(
+  config = ConfigFactory.parseString("""{
+    aws {
+      access-key = "<access-key>"
+      secret-key = "<secret-key>"
+    }
+  }"""),
+  accessKeyPath = "aws.access-key",
+  secretKeyPath = "aws.secret-key")
+
+val credentials = confProvider.getCredentials
```

Comment:
This section should probably be behind a `silent` block. We're not interested in showing the output and `credentials` doesn't have a stable `toString`.